### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by Caren-Koroeny**
<h1>Hey There 👋🏻</h1>

> In this PR, I:
<details>
  <summary>Interface</summary>
  <ul>
    <li> 
- [ ] Create a class `Nameable`. 

- [ ] Implement a method called `correct_name` that will raise a `NotImplementedError`.

- [ ] Turn your Person class to Nameable

- [ ]  Make sure that your Person class inherits from Nameable - Make sure that this class has a method `correct_name` implemented. It should simply return the name attribute.</li>
  </ul>
</details>

<details>
<summary>Prepare base Decorator</summary>
<ul>
<li>

- [ ] Make sure that it inherits from Nameable. 

- [ ] In the constructor assign a nameable object from params to an instance variable. 

- [ ] Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.</li></ul>

</details>

<details>
<summary>Prepare CapitalizeDecorator and TrimmerDecorator</summary>
<ul>
<li>

<h3> For the CapitalizeDecorator:</h3>

- [ ]  Create a class that inherits from the base Decorator class.
 
- [ ]  Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`. 

<h3> For the TrimmerDecorator:</h3>

- [ ]  Create a class that inherits from the base Decorator class. 

- [ ]  Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

</li>
</ul>
</details>
<h2>Kindly Review</h2>

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
+require_relative './decorators/nameable'
 class Person
```
Comment (franclobo) [inline person.rb]: - [ ] Person class should inherits from Nameable, please 🙏🏽 use inheritance. We use inheritance as a way to extract common behaviors from classes that share that behavior and move it to a superclass. This lets us keep logic in one place.

File: decorators/capitalize_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require_relative './decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.CapitalizeDecorator
```
Comment (franclobo) [inline decorators/capitalize_decorator.rb]: - [ ] CapitalizeDecorator method is not defined. May you want to say `capitalize`? Capitalize method capitalize the first letter of the given name.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @Caren-Koroeny, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️
- OOP ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 3]
```diff
@@ -1,5 +1,7 @@
-class Person
+require_relative './decorators/nameable'
```
Comment (rotimiazeez) [inline person.rb:1]: Your `person` class should require the capitalize_decorator and trimmer_decorator at the top to ensure the below error is fixed
Screenshot from 2023-05-17 15-03-59
Adding this should fix it
```
require_relative './decorators/capitalize_decorator'
require_relative './decorators/trimmer_decorator'
```

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
Hi @Caren-Koroeny 😉, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlight
- Good job implementing the changes required by the previous CR

## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

**Review by christianbiring1 — APPROVED** [review]
Hi @Caren-Koroeny , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
# STATUS: APPROVED :tada: 

## Highlights
- [ ] All linter checks are green :heavy_check_mark: 
- [ ] Descriptive PR `title` and `summary` :+1: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**